internal/blocksync/reactor.go

```diff
@@ -490,9 +519,12 @@ func (r *Reactor) poolRoutine(ctx context.Context, stateSynced bool, blockSyncCh
 			// TODO: Uncouple from request routine.
 
 			// see if there are any blocks to sync
-			first, second := r.pool.PeekTwoBlocks()
-			if first == nil || second == nil {
-				// we need both to sync the first block
+			first, second, extCommit := r.pool.PeekTwoBlocks()
+			if first == nil || second == nil || extCommit == nil {
+				if first != nil && extCommit == nil {
+					r.logger.Error("peeked a block without extended commit", "height", first.Height)
```

Comment:
is this really an error?

Reply:
It is in the current version of the solution. A received block in block sync interactions must _always_ contain an extended commit.

Reply:
Given that we should always have extended commits, should there not be some consequence somewhere for this? Or is logging the issue sufficient? I suppose it makes sense to have this as informational if some other routine will pick up on the fact that the relevant extended commit is missing and panic/error out.

Reply:
If we hit this line, there is obviously a bug somewhere else in this reactor (probably in `PeekTwoBlocks`). I decided to just log an error an not panic because we can recover from this, but I wouldn't be against a panic since the values we get from `PeekTwoBlocks` are totally under our control.

Reply:
Technically, not having the first or the second is not an error (we simply might not have yet received a block at that height). However, having a block and no extended commit is an error. PeekTwoBlocks is supposed to just read whatever is in the store. So if there is an error here, it should be in incorrectly stored blocks, not a bug. What will happen here is still ok, we will try to read the same block again. But I also think that, if we end up in this situation, there is something wrong with Tendermint or there are inconsistencies in the data, no?

Reply:
I've updated this to panic instead of just logging the error: https://github.com/tendermint/tendermint/pull/8433/commits/9483b42f0b5a1362f2b6f096e26149cf4370974f